docs/autogen/dest.py

```diff
@@ -105,6 +114,43 @@ def remove_except(path, keep):
                 shutil.rmtree(item)
 
 
+def gen_versions(path: Path, current: str):
+    """Returns a dict compatible with the Pydata versions-format"""
+
+    def version_key(name):
+        match = re.match(r"v(\d+)\.(\d+)\.(\d+)", name)
+        if match:
+            return (1, -int(match.group(1)), -int(match.group(2)), -int(match.group(3)))
+        return (0, name)
+
+    versions = [
+        {
+            "name": "current",
+            "version": f"{current}",
+            "url": "https://xnvme.org/",
+            "preferred": True,
+        },
+    ]
+
+    for item in sorted(path.iterdir(), key=lambda x: version_key(x.name)):
+        versions.append(
+            {
+                "name": f"{item.name}",
+                "version": f"{item.name}",
+                "url": f"/docs/{item.name}",
+            }
+        )
```

Comment:
Instead of having to write `.name` four times you could do a list comprehension inside the sorted call.
Something like: `[x.name for x in path.iterdir()]`.
That would also allow you to get rid of the lambda: `key=version_key`

Reply:
Would this be preferable:
```python
    for vname in (
        x.name for x in sorted(path.iterdir(), key=lambda x: version_key(x.name))
    ):
        versions.append(
            {
                "name": f"{vname}",
                "version": f"{vname}",
                "url": f"/docs/{vname}",
            }
        )
```

I don't see how skipping ``key=version_key`` is possible without losing the ordering?

Reply:
Or something like:

```python
    for dname in sorted(
        (d.name for d in path.iterdir()), key=lambda x: version_key(x.name)
    ):
        versions.append(
            {
                "name": f"{dname}",
                "version": f"{dname}",
                "url": f"/docs/{dname}",
            }
        )
```

Reply:
This is what I mean:
```python
for name in sorted((d.name for d in path.iterdir()), key=version_key):
        versions.append(
            {
                "name": f"{name}",
                "version": f"{name}",
                "url": f"/docs/{name}",
            }
        )
```

Reply:
Nice, adjusted it to this.